### ISSUE TITLE
figure out which forms are possible in which roles

### ISSUE THREAD

**Issue by jonorthwash** [apertium/apertium-sah#5]
Which of these forms can be used in which ways?

| suffix form | example forms | finite? | subst/ger? | advl/gna? | attr/gpr? | with aux / prc? |
|---------------|----------------------|---|---|---|---|---|
| -{А}р | барар, саныыр | yes |  |  |  |  |
| -{B}{I}т | барбыт, санабыт | yes |  |  |  |  |
| -{I}{A}х | барыах, буолуох | yes |  |  |  |  |
| -{I}{A} | барыа, буолуо |  |  |  |  |  |
| -{D}{A}х | бардах, бүтэрдэх | yes |  |  |  |  |
| -{A}{A}чч{I} | барааччы, үүнээччи | yes |  |  |  |  |
| -{A} илик | бара илик, үөрэнэ илик | yes |  |  | yes |  |
| -{A}н | баран, киирэн |  |  |  |  |  |
| -{A} | бара, ыы |  |  |  |  |  |
| -{I}м{I}н{A} | барымына, ылымына | yes |  |  |  |  |
| -{B}{A}кк{A} | барбакка, тэһийбэккэ | yes? |  |  |  |  |
| -{а}{а}р{I} | бараары, диэри |  |  | yes |  |  |

**Comment by varie** [comment]
Yeah, sure! Could you give some examples, please?

**Comment by jonorthwash** [comment]
Okay, some initial guidelines:

Verbal nouns (=gerunds) are forms that can do any of the following:
- act as the subject of a sentence, 
- take various case suffixes and act in other roles in a sentence that a noun or pronoun might normally occur,
- take possessive suffixes, though not necessarily in the meaning of possession (e.g., maybe to agree with the subject of the verb).

Try filling in that column first, and note any limitations (like I did for -{I}{A}n).  I'll provide more guidelines later, maybe on IRC.

**Comment by jonorthwash** [comment]
Verbal adjectives (attr/gpr = глагольное прилагательное) are verb forms that can modify nouns in much the same way as adjectives.  For example, бара илик киһи,

**Comment by jonorthwash** [comment]
Verbal adverbs (`advl/``gna` = глагольное наречие) and `prc` (≈participle/причастие) are very similar, so this can be difficult.  They both occur with other verbs, but the difference is that verbal adverbs [usually] form two predicates ("He ate _as he was fishing_") whereas `prc` occurs only with auxiliaries to create a single predicate ("He can _fish_", or "He finished _fishing_").  Often in Turkic languages, verbal nouns plus various case markers have similar uses and meanings to verbal adverb form, but they should be counted as verbal noun + case if they are productive.

**Comment by jonorthwash** [comment]
Here are some Sakha examples of `prc` that I found in lexc comments:
_сиэн_ кэбис
_бара_ сыста

**Comment by varie** [comment]
Видообразующие служебные глаголы combine with -ан and -а (ыы) forms and become grammatical indicator of an aspect of a verb. Ubryatova describes following groups of видообразующие служебные глаголы:
1. эр;
2. оҕус, кэбис, хаал;
3. ис, тур, түс;
4. бар, сырыт.

Модальные служебные глаголы have modal and modal-temporal meaning. Ubryatova desribes following groups:
1. Main group is functional modal verbs: баҕар, сатан, табылын, саараа, куттан, кыбыһын, салын, толун, санаа, саат, аһын, туттун, дьулай, сыыс, уолдьас and others;
2. буол, сыс, этэ, көр, сатаа and others.

**Comment by jonorthwash** [comment]
> Модальные служебные глаголы

What verb forms do these combine with?

And could you give some prc examples of verb forms like барар and барыах? 

Also, you mentioned one other type of modal the other day?

**Comment by jonorthwash** [comment]
There are some Kazakh examples of gna on the Apertium wiki here: http://wiki.apertium.org/wiki/Turkic_lexicon#Verbal_adverbs

Does this help clarify?

**Comment by varie** [comment]
> Модальные служебные глаголы
What verb forms do these combine with?
1. First group combine with причастие будущего времени в форме винительного и дательного падежей (-ыаҕын или -ыахха + modal verb). 
Examples:
Ааҕаччы болҕомтотун биир түбэлтэҕэ _тохтотуохпун баҕарабын_. - I would like to draw the reader's attention to one occasion.
Вася, мин эмиэ партияҕа _киириэхпин саныыбын_. - Vasya, I also think about joining the party
2. Second group combine with:

- present -ар + буол
Арыт киэһэлэргэ бэрэбинэлэр үрүлэригэр тахсан хамсаабакка бүк түһэн _олорор буолар_ - In other evenings sometimes he sits hunched over on logs.
Аҕыйах хонон баран Михей куоракка үөрэнэ _киирэр буолар_ - A few days later, Micah decides to go to town to study.

- past -быт + буол
Семену көрөөт, _соһуйбута буолллулар_ - When they saw Semyon, they pretended to be surprised.

- future -ыах + буол
«Не верь, не бойся, не проси!» диэн ырыанан _барыах буолбута_ - They announced that they will go with song «Не верь, не бойся, не проси!»

- -а, -ыы + сыс
Ол ыккардыгар хаста да ууга _бара сыстылар_ - Meanwhile, they nearly drowned several times.

- -уох + этэ
Оччоҕо бэйэбит туспа аҕабыыттаах _буолуох этибит_ - Then we ourselves separately would have our priest.

- -ан + көр
Арай бу эмээхсинтэн _көрдөөн көрүөҥ_ буолуо - Hardly you would try to ask this old lady.

- -а, -ыы +  сатаа
_Оҥоро сатаата_ да, туох да тахсыбата - He tried to do it, but he did not succeed.

All examples are from Ubryatova book except 'future -ыах + буол'. 'future -ыах + буол' example is from Wikipedia corpus.

**Comment by jonorthwash** [comment]
I don't understand what the first and second groups are... Those are all with модальные служебные глаголы?

Also, what about my other questions?

**Comment by jonorthwash** [comment]
> First group combine with причастие будущего времени в форме винительного и дательного падежей (-ыаҕын или -ыахха + modal verb).

The two examples of this both appear to be -ыахпын.  Could you explain how this corresponds to the the explanation?

**Comment by varie** [comment]
> I don't understand what the first and second groups are... Those are all with модальные служебные глаголы?

Yes, Ubryatova divided them into two groups: functional modal verbs and for second group she didn't give any specific name.

> And could you give some prc examples of verb forms like барар and барыах?

Are these fine?
Арыт киэһэлэргэ бэрэбинэлэр үрүлэригэр тахсан хамсаабакка бүк түһэн _олорор буолар_ - In other evenings sometimes he sits hunched over on logs.
Аҕыйах хонон баран Михей куоракка үөрэнэ _киирэр буолар_ - A few days later, Micah decides to go to town to study.
«Не верь, не бойся, не проси!» диэн ырыанан _барыах буолбута_ - They announced that they will go with song «Не верь, не бойся, не проси!»

> Also, you mentioned one other type of modal the other day?

I don't remember :(.

> There are some Kazakh examples of gna on the Apertium wiki here: http://wiki.apertium.org/wiki/Turkic_lexicon#Verbal_adverbs
Does this help clarify?

Yes, thanks! I figured out that verbal adverb is деепричастие and filled the chart according to list of  деепричастие from Ubryatova book. 

> First group combine with причастие будущего времени в форме винительного и дательного падежей (-ыаҕын или -ыахха + modal verb).
The two examples of this both appear to be -ыахпын. Could you explain how this corresponds to the the description?

Hmm, might it be 1 person possessive singular accusative case of -ыах form?

1p: барыахпын баҕарабын
2p: барыаххын баҕараҕын
3p: барыаҕын баҕарар

And then there are also several other examples with functional modal verbs (first group):
Суох, Федор Васильевич, _сэргэҕэлээтэххинэ сатанар_ - No, Fyodor Vasilyevich, you need to be interested.
Тогойкин уокка _чугаһыан дьулуйда_ - Togoikin did not dare to approach the fire.
Кэбис, бу дойдуттан дьалты буола _оҕустахха сатаныыһы_ - No, from this place (we) need to get away soon. - ah, is it -ыахха here?
Онон илин өттүбүтүгэр Япония баарын _умнуохпутун сатаммат_ - Therefore, we must not forget that on our eastern side there is Japan.
Таптыыр киһитин кытта _иирсиэн_, _айдаарсыан кэрэйдэ_ - He refrained from quarreling, arguing with a loved one.
Эйиэхэ _биэриэхпин саарбахтыыбын_ ээ - I doubt to give it to you.

So, they seem to combine with other forms also. Those (-ыаҕын или -ыахха + modal verb) were some particular examples for accusative and dative cases of -ыах form (p. 406 of the book)..

**Comment by jonorthwash** [comment]
> I figured out that verbal adverb is деепричастие and filled the chart according to list of деепричастие from Ubryatova book.

Verbal adverbs and `prc` are both called деепричастие in Russian.  It's important to be able to keep them separate

> Hmm, might it be 1 person possessive singular accusative case of -ыах form?
>
> 1p: барыахпын баҕарабын
> 2p: барыаххын баҕараҕын
> 3p: барыаҕын баҕарар

These are accusative??  Aren't these just finite forms?  That is, can't they be the main verb at the end of a sentence / the only verb in a sentence?
.

**Comment by varie** [comment]
> Verbal adverbs and prc are both called деепричастие in Russian. It's important to be able to keep them separate

Aa! I will recheck advl/gna column.

> These are accusative?? Aren't these just finite forms? That is, can't they be the main verb at the end of a sentence / the only verb in a sentence?

Hmm.. Ubryatova says that причастие на -ыах gets possessive affixes and makes indicative future tense:
барыаҕым
барыаҕыҥ
барыаҕа
барыахпыт
барыаххыт
барыахтара

So then accusative case of those forms would be:
барыахпын
барыаххын
барыаҕын
барыахпытын
барыаххытын
барыахтарын

I think they are not finite forms. I found only these three occurrences in Wikipedia corpus: 
„Хаһан кэлэллэрий? Туох эрэ буолан эрэбин, балыыһаҕа _барыахпын_ наада“, — диир.
Ол гынан баран правительство онно харчы _барыаҕын_ баҕарбатаҕа уонна Соҕуруу Африка улахан өттүн баай өттүн маҥан дьоҥҥо хаалларыан баҕарара.
Бу дойдуга гражданскай сэрии олус уһун кэмҥэ уонна ыардык _барыаҕын_ Аан дойду демократияны тутуһар дьонноро элбэх буоланнар көмөлөөн, мөрөйдөөн тохтоппутара.

**Comment by jonorthwash** [comment]
Aha, this is making more sense to me now.  In this case, -IAх is a verbal noun suffix.

I also note that in the second group, - IAх, -Aр, and -BIт only occur with буол.  This is good evidence that they are acting as verbal nouns (in nominative) in those contexts.

Evidence that would suggest these are verbal nouns would include uses with буол where the subject of the verb and буол is different.  Can the verb in one of these forms be e.g. first person, while буол is e.g. 3rd person?

Evidence that would suggest these are `prc` would include uses of these forms with other auxiliary verbs. Can you find examples of that?

**Comment by jonorthwash** [comment]
From IRC:

"Мин куоракка киирэрим буолар." - I'll decide to go to the city.
"Мин куоракка киирэр буолабын." - Me going to the city happens.

**Comment by jonorthwash** [comment]
Forms like "буоллаҕына" we had as `<gna_cond2>`, but it appears they're better analysed as just `<ger_cond><px3sg><par>`.

**Comment by jonorthwash** [comment]
@varie, could you give some examples of -ыа (барыа, буолуо, саныа)?  I think it's probably `<gna_fut>` or so (as it's currently in the transducer), but the table says it's `<ger>`...?

**Comment by jonorthwash** [comment]
> Тогойкин уокка _чугаһыан дьулуйда_ - Togoikin did not dare to approach the fire.
> Таптыыр киһитин кытта иирсиэн, _айдаарсыан кэрэйдэ_ - He refrained from quarreling, arguing with a loved one.

Based on these examples, I think `-{I}{A}н` makes `<prc>`, not `<ger>`.  @varie, what do you think?  Do _дьулуйда_ or _кэрэйдэ_ occur with nouns too?  What about with other verb forms?

**Comment by varie** [comment]
@jonorthwash, I found these examples from the newspapers corpus (translations might be not very accurate, but I tried my best :):
Боростуой киһи сылдьар кыахтаах эрэ буоллаҕына билиэттэрэ _атыыга барыа_ буоллаҕа эбээт. 
If common people can go there, only then their tickets _will be sold_.
Бука, Алампа да элбэх кумааҕыта уматылыннаҕа _буолуо ээ_.
_It's probable_ that even Alampa's many papers were burnt.
Буруйдаммыттар _хом саныа_ суохтаахтар.
Accused ones _should not feel sad_.

**Comment by varie** [comment]
@jonorthwash yeah, I think that -{I}{A}н is not `<ger>`. I think that чугаһыан or иирсиэн cannot be a subject of a sentence. 
Do дьулуйда or кэрэйдэ occur with nouns too? - yes, for example: "харчытын кэрэйдэ" - he didn't want to spend his money.
What about with other verb forms? - yes, for example: "ыраах барыахпын кэрэйдим" - I didn't dare to go far

**Comment by jonorthwash** [comment]
> Do дьулуйда or кэрэйдэ occur with nouns too?
>> - yes, for example: "харчытын кэрэйдэ" - he didn't want to spend his money.
>
> What about with other verb forms?
>> - yes, for example: "ыраах барыахпын кэрэйдим" - I didn't dare to go far

These are both evidence, in my mind, of `-{I}{A}н` being a `<ger>`.  Note that a form doesn't have to be able to be the subject of a sentence to be a gerund.

> I found these examples from the newspapers corpus

Those seem to be examples of `-{I}{A}`, not `-{I}{A}н`?

**Comment by jonorthwash** [comment]
@varie, could you give a couple examples of each of the last three types in the table (`{I}м{I}н{A}`, `{B}{A}кк{A}`, `{a}{a}р{I}`), both non-finite, and if present, finite forms?

**Comment by varie** [comment]
There are examples from Ubryatova's book:
- `{I}м{I}н{A}`
Биллиҥ? - Билиминэ. - Did you recognize? - _Of course I recognized_.
Мааҕыан, Бойуот барыах диэбитигэр, баран _хаалымынабын_. - Just recently, when Boyot invited me, I _should have gone_.

- `{B}{A}кк{A}`
"Кыайар баара сыттыкпын _аҕалбаккабын_." I _didn’t take_ the pillow with me _in vain_, though I could have taken it.
"Эдьигээҥҥэ эмиэ дьиэ-уот _тиийбэккэ_, элбэх ыал кыараҕастык олорор." As there _is not enough_ housing in Zhigansk, many people live in cramped conditions.

- `{a}{a}р{I}`
"Арҕаа Европаны эргийэ _айаннаары_ кэллибит." We came _to travel_ around Western Europe.
"Ыһыахха _сылдьаарыбын_ дойдулаатым." I went to my native area _to visit_ Ysyakh (Midsummer celebration).

**Comment by jonorthwash** [comment]
@varie, very helpful, thanks!  Can you find some non-finite forms of `{I}м{I}н{A}`?

**Comment by jonorthwash** [comment]
There appears to be some possibility that `{a}{a}р{I}` could be `prc` in addition to `gna`, but we'd have to say бэлэмнэн is an auxiliary.  This is something to consider more deeply in the future.

**Comment by jonorthwash** [comment]
The following forms are no longer being analysed:
- буолуохтарын, көрдөрүмүөхтэрин, кэлиэхтэрин
  - Are these finite future forms (e.g., `буол<v><iv><fut><p3><pl>`) or non-finite forms (e.g., `буол<v><iv><ger_fut><px3pl><acc>`) or what?
- диэххэ, туттуохха, булуохха, көрүөххэ, бэлиэтиэххэ, өйдүөххэ, туһаныахха, оҥоруохха, ааттыахха, быһаарыахха, билиэххэ, ыйыахха, этиэххэ, ылыахха
  - Are these `<ger_fut><dat>`?
- буолуоҕуттан
  - Is this `<ger_fut><px3sg><abl>`?
- өрүтүн, устубут
  - I believe these are both forms of postposition-like nouns, right?

**Comment by jonorthwash** [comment]
Also, @varie, which forms are right here?  I.e., is it барбатах or барбабыт, or are both right (with different meanings or not?)?
```
[5/6][FAIL] бар<v><iv><neg><gpr_past> => Missing results: барбатах
[5/6][FAIL] бар<v><iv><neg><gpr_past> => Unexpected results: барбабыт
[6/6][FAIL] санаа<v><tv><neg><gpr_past> => Missing results: санаабатах
[6/6][FAIL] санаа<v><tv><neg><gpr_past> => Unexpected results: санаабабыт
```

**Comment by varie** [comment]
- өрүтүн, устубут, өрүтү, өрүтүнэн
I think these are just form of өрүт noun. I am not sure if they are used as postpositions.
устубут is a form of past singular 3p уһул and үһун verbs.

- истэллэр
form of present plural 3p иһит verb

- right:
```
[5/6][FAIL] бар<v><iv><neg><gpr_past> => Missing results: барбатах
[6/6][FAIL] санаа<v><tv><neg><gpr_past> => Missing results: санаабатах
```

- not right:
```
[5/6][FAIL] бар<v><iv><neg><gpr_past> => Unexpected results: барбабыт
[6/6][FAIL] санаа<v><tv><neg><gpr_past> => Unexpected results: санаабабыт
```

**Comment by jonorthwash** [comment]
Related: #25.

**Comment by jonorthwash** [comment]
I will finish going through the table and making sure it's all in the `lexc` file correctly.
